src/Piece/Pawn.cpp

```diff
@@ -0,0 +1,24 @@
+#pragma once
+#include "Piece/Pawn.h"
+
+bool Pawn::isValidMove(const Position src, const Position des) const {
+    if(isFirstMove)
+        if(this->isWhite())
+            return (des.row == src.row + 2 || des.row == src.row + 1) && des.col == src.col;
+        else
+            return (des.row == src.row - 2 || des.row == src.row - 1) && des.col == src.col;
+    else
+        if(this->isWhite())
+            return des.row == src.row + 1 && des.col == src.col;
+        else
+            return des.row == src.row - 1 && des.col == src.col;
```

Comment:
Use {} for if-else blocks: It's considered good practice to use braces {} for if-else blocks, even if they consist of only a single statement. This helps avoid potential bugs and improves code readability.